Bottom: 7eeb1cbcd221991e5aacca89a3e1f949395ed57e
Top:    92b2f26dfeb828d5a7ce5f72fd9b19d3ae9e0818
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-22 00:30:35 +0100

Add profiling to file load







---

diff --git a/src/file.c b/src/file.c
index 738188a..35bfb57 100644
--- a/src/file.c
+++ b/src/file.c
@@ -350,6 +350,10 @@ LoadPCB (char *Filename)
 {
   PCBTypePtr newPCB = CreateNewPCB (False);
   Boolean units_mm;
+  clock_t start, end;
+  double elapsed;
+
+  start = clock();
 
   /* new data isn't added to the undo list */
   if (!ParsePCB (newPCB, Filename))
@@ -380,6 +384,10 @@ LoadPCB (char *Filename)
 	  CreateDefaultFont ();
 	}
 
+  end = clock();
+  elapsed = ((double) (end - start)) / CLOCKS_PER_SEC;
+  printf ("Loading file %s took %f\n", Filename, elapsed);
+
       /* clear 'changed flag' */
       SetChangedFlag (False);
       PCB->Filename = MyStrdup (Filename, "LoadPCB()");
